github/commitstatus.go

```diff
@@ -69,14 +74,18 @@ type CommitStatus struct {
 //
 // See also:
 // - https://docs.github.com/en/rest/commits/statuses
-func NewCommitStatus(target *Target, token, owner, repo, context string, log *slog.Logger) CommitStatus {
+func NewCommitStatus(target *Target, token, owner, repo, context string, log *slog.Logger, useGithubAppAuth bool, privateKey string, applicationId, installationId int64) CommitStatus {
```

Comment:
I would like to avoid having so many function arguments...

Reply:
i thought about that too, we could have NewCommitStatusGithubApp and NewCommitStatusAccessToken i suppose? other than that, I am not sure of how to avoid the issue

Reply:
I have some ideas, but need to first understand in detail what `github.com/bradleyfalzon/ghinstallation` does...